modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProductImages.macro

```diff
@@ -1,5 +1,70 @@
 definition {
 
+	macro addAttachmentImages {
+		for (var imageTitle : list "${imageTitles}") {
+			echo("imageTitle is equal to ${imageTitle}");
+
+			var image = StringUtil.extractFirst("${imageTitle}", ":");
+
+			var number = StringUtil.extractLast("${imageTitle}", ":");
+
+			var count = "0";
+
+			if ("${image}" == "") {
+				image = "${imageTitle}";
+			}
+
+			var uploadFileName = "Commerce_${image}.jpg";
+
+			Refresh();
+
+			Click(locator1 = "CommerceEntry#ATTACHMENT_MODAL_IFRAME");
+
+			Click(locator1 = "CommerceEntry#ATTACHMENT_PLUS_BUTTON");
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			Click(locator1 = "Button#SELECT_FILE");
+
+			SelectFrameTop();
+
+			CommerceItemSelector.uploadSingleFile(uploadFileName = "${uploadFileName}");
+
+			SelectFrameTop();
+
+			Button.clickAdd();
+
+			if (isSet(optionValues)) {
+				var optionValue = StringUtil.extractFirst("${optionValues}", ",");
+
+				if ("${optionValue}" == "") {
+					optionValue = "${optionValues}";
+				}
+
+				CommerceEntry.selectAssociatedOptionsValue(optionValue = "${optionValue}");
+
+				var optionValues = StringUtil.replaceFirst("${optionValues}", "${optionValue},", "");
+			}
+
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+
+			CommerceEntry.publishProductImage();
+
+			Click(locator1 = "Icon#CLOSE");
+
+			Refresh();
+
+			if (isSet(validationImage)) {
+				CommerceEntry.checkImageName(
+					image = "${image}",
+					number = "${number}",
+					uploadFileName = "${uploadFileName}");
+			}
+		}
+
+		Refresh();
+	}
+
```

Comment:
This macro looks like that was copy/pasted from this other PR … so, please, cherrypick the code that the macro was created, to avoid the copy/paste way, as it can cause a rebase problems

Reply:
Ok, we created this macro for the two tests (6030 and 6031), so it was copied / pasted. I couldn't use the macro because it didn't exist yet. How should I proceed in this case?